### ISSUE TITLE
Dispatcher not found

### ISSUE THREAD

**Issue by voelklmichael** [dotnet/wpf#5852]
tl;dr  How to solve "Reference to type 'DispatcherObject' claims it is defined in 'WindowsBase', but it could not be found"

Hi,
so I have this app which does the following:
- takes some user-defined source code snippets
- compiles it (using rosyln)
- and executes it.
Yesterday a user wanted to run some source code which opens a window.
The source code uses "System.Windows.Application.Current.Dispatcher.Invoke(() =>{…});".
Now I get an error during compilation of this source snippet:
    ID CS7069: Reference to type 'DispatcherObject' claims it is defined in 'WindowsBase', but it could not be found
    ID: CS1061: 'Application' does not contain a definition for 'Dispatcher' and no accessible extension method 'Dispatcher' accepting a first argument of type 'Application' could be found (are you missing a using directive or an assembly reference?)
I'm tried to add several DLL references from "C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App\3.1.21", but no success.

So: In which DLLs is the Dispatcher defined?

Note: If I run the snippet in a minimal app, Visual Studio chooses the references automatically, and the code just works.
On stackoverflow I found some issues, which indicate that I've already loaded another version of "WindowsBase". Do you know how I can check this?

Thank you,

Michael

Update: I found the issue. I loaded WindowsBase from "Microsoft.NETCore.App" instead of "Microsoft.WindowsDesktop.App".
Please close this issue.

**Comment by lindexi** [comment]
Could you replace the code to your csproj file?

```
<Project Sdk="Microsoft.NET.Sdk.WindowsDesktop">

  <PropertyGroup>
    <OutputType>WinExe</OutputType>
    <TargetFrameworks>netcoreapp3.1;net5.0-windows</TargetFrameworks>
    <UseWPF>true</UseWPF>
    <Nullable>enable</Nullable>

    <IsPackable>false</IsPackable>
  </PropertyGroup> 


</Project>
```
